core/network/impl/peer_manager_impl.cpp

```diff
@@ -72,6 +72,7 @@ namespace kagome::network {
 
     // Add themselves into peer routing
     kademlia_->addPeer(host_.getPeerInfo(), true);
+    processDiscoveredPeer(host_.getPeerInfo().id);
```

Comment:
This is needless. We never connect to themselves over network, only LoopbackStream

Reply:
It starts protocols for dev-mode, see `PeerManagerImpl::isSelfPeer`